pkgs/applications/networking/instant-messengers/rambox/pro.nix

```
@@ -0,0 +1,46 @@
+{ autoPatchelfHook, electron, fetchurl, makeDesktopItem, makeWrapper, nodePackages, nss, stdenv, xdg_utils, xorg }:
+
+stdenv.mkDerivation rec {
+  pname = "rambox-pro";
+  version = "1.1.2";
+
+  dontBuild = true;
+  dontStrip = true;
+
+  buildInputs = [ nss xorg.libxkbfile ];
+  nativeBuildInputs = [ autoPatchelfHook makeWrapper nodePackages.asar ];
+
+  src = fetchurl {
+    url = "https://github.com/ramboxapp/download/releases/download/v${version}/RamboxPro-${version}-linux-x64.tar.gz";
+    sha256 = "0rrfpl371hp278b02b9b6745ax29yrdfmxrmkxv6d158jzlv0dlr";
+  };
+
+  patches = [ ./fix-bash-args.patch ];
```

```suggestion
  postPatch = ''
    substituteInPlace resources/app.asar.unpacked/node_modules/ad-block/vendor/depot_tools/create-chromium-git-src \
      --replace "/usr/bin/env -S bash -e" "${stdenv.shell}"
    substituteInPlace resources/app.asar.unpacked/node_modules/ad-block/node_modules/bloom-filter-cpp/vendor/depot_tools/create-chromium-git-src \
      --replace "/usr/bin/env -S bash -e" "${stdenv.shell}"
  '';
```
is what I could come up with. This is not totally a blocker though.

This hotfixing does not carry the `set -e` of your patch, but I could build and run the app properly without it (on NixOS). Is it really necessary? Maybe on Darwin?

---

Looks sensible - nope - I see no reason to add unnecessarily complexity if it can just be patched out with a string replace.

Looks good! 👍